server/route.go

```diff
@@ -637,11 +641,26 @@ func (s *Server) routeAcceptLoop(ch chan struct{}) {
 
 	// Check for TLSConfig
 	tlsReq := opts.Cluster.TLSConfig != nil
+	// Configure Cluster Advertise Address
+	host := opts.Cluster.Host
+	port = l.Addr().(*net.TCPAddr).Port
+	ip := ""
+	if opts.Cluster.RouteAdvertise != "" {
+		advHost, advPort, err := parseHostPort(opts.Cluster.RouteAdvertise, strconv.Itoa(port))
```

Comment:
In both places where you use parseHostPort, it seems that you have access to port as an integer. Not sure why you pass it as a string. I would rather have parseHostPort convert port to string only when it gets an error and need to construct an address (as string). See comments in that function.